scripts/build_test.py

```diff
@@ -14,29 +14,270 @@
+        # Reading from actual base.html file and not simply a dummy file.
+        build._ensure_files_exist([base_source_path])
+        # Assert that base.html has white spaces.
+
+        with open(base_source_path, 'r') as source_base_file:
+            source_base_file_content = source_base_file.read()
+            self.assertRegexpMatches(
+                source_base_file_content, r'\s{2,}',
+                msg="No white spaces detected in %s unexpectedly"
+                % base_source_path)
+            # Create a list of un-hashed filenames in base.html.
+            base_file_unhashed_filenames = []
+            for filepath, _ in file_hashes.iteritems():
+                # Not hashing HTML files.
+                if filepath.endswith('.html'):
+                    continue
+                result = re.findall(filepath, source_base_file_content)
+                if result:
+                    base_file_unhashed_filenames.append(result[0])
+            self.assertGreater(len(base_file_unhashed_filenames), 0)
+
+            build.process_html(
+                source_base_file, minified_html_file_stream, file_hashes)
+
+        minified_html_file_content = minified_html_file_stream.getvalue()
+        self.assertNotRegexpMatches(
+            minified_html_file_content, r'\s{2,}',
+            msg='All white spaces must be removed from %s' %
+            base_source_path)
+        # Assert that all filenames must be hashed.
+        for unhashed_filename in base_file_unhashed_filenames:
+            result = re.findall(
+                unhashed_filename, minified_html_file_content)
+            self.assertEqual(len(result), 0)
+
+    def test_hash_should_be_inserted(self):
+        """Test hash_should_be_inserted to return the correct boolean value
+        for filepath that should be hashed.
+        """
+        with self.swap(
+            build, 'FILEPATHS_NOT_TO_RENAME', (
+                'path/to/fonts/*', 'path/to/third_party.min.js.map',
+                'path/to/third_party.min.css.map')):
+            self.assertFalse(build.hash_should_be_inserted(
+                'path/to/fonts/fontawesome-webfont.svg'))
+            self.assertFalse(build.hash_should_be_inserted(
+                'path/to/third_party.min.css.map'))
+            self.assertFalse(build.hash_should_be_inserted(
+                'path/to/third_party.min.js.map'))
+            self.assertTrue(build.hash_should_be_inserted(
+                'path/to/wrongFonts/fonta.eot'))
+            self.assertTrue(build.hash_should_be_inserted(
+                'rich_text_components/Video/protractor.js'))
+
+    def test_generate_copy_tasks_to_copy_from_source_to_target(self):
+        """Test generate_copy_tasks_to_copy_from_source_to_target to queue up
+        the same number of copy tasks as the number of files in the directory.
+        """
+        assets_hashes = build.get_file_hashes(build.ASSETS_DEV_DIR)
```

Comment:
Same note re using a test folder.

Ditto in multiple places below.

Reply:
Done!